### ISSUE TITLE
bug in eval.detect ?

### ISSUE THREAD

**Issue by MrKamiZhou** [argman/EAST#214]
Thanks for your great work! But there might be some problems...

https://github.com/argman/EAST/blob/ab97939783901b7e22ff55e151964e159d1627b9/eval.py#L89

` text_box_restored = restore_rectangle(xy_text[:, ::-1]*4, geo_map[xy_text[:, 0], xy_text[:, 1], :])` 
will return coordinates of each bbox with different order, because the function 'restore_rectangle' handles pos and neg angles separately.

But the following codes 
` boxes = np.zeros((text_box_restored.shape[0], 9), dtype=np.float32)
    boxes[:, :8] = text_box_restored.reshape((-1, 8))
    boxes[:, 8] = score_map[xy_text[:, 0], xy_text[:, 1]]` 
don't seem to take that into consideration...
So the nms part will process with a wrong score for each bbox

Am I right?

@argman

**Comment by MrKamiZhou** [comment]
`def restore_rectangle_v1(origin, geometry):
    d = geometry[:, :4]
    angle = geometry[:, 4]
    coordinates = np.hstack(np.hstack([-d[:,3:],d[:,0:1],d[:,1:2],d[:,0:1],d[:,1:2],-d[:,2:3],-d[:,3:],-d[:,2:3]])).reshape(-1,4,2).transpose(1,0,2)
    rotateX = np.array([-np.sin(angle),-np.cos(angle)]).T
    rotateY = np.array([-np.cos(angle),-np.sin(angle)]).T
    X = (coordinates*rotateX).sum(axis=2) + origin[:,1]
    Y = (coordinates*rotateY).sum(axis=2) - origin[:,0]
    result = np.concatenate((-Y.reshape(4,-1,1),X.reshape(4,-1,1)),axis=2).transpose(1,0,2)
    return result`
can solve this problem

**Comment by MrKamiZhou** [comment]
`def restore_rectangle_v1(origin, geometry):
    d = geometry[:, :4]
    angle = geometry[:, 4]
    coordinates = np.hstack(np.hstack([-d[:,3:],d[:,0:1],d[:,1:2],d[:,0:1],d[:,1:2],-d[:,2:3],-d[:,3:],-d[:,2:3]])).reshape(-1,4,2).transpose(1,0,2)
    rotateX = np.array([-np.sin(angle),-np.cos(angle)]).T
    rotateY = np.array([-np.cos(angle),-np.sin(angle)]).T
    X = (coordinates*rotateX).sum(axis=2) + origin[:,1]
    Y = (coordinates*rotateY).sum(axis=2) - origin[:,0]
    result = np.concatenate((-Y.reshape(4,-1,1),X.reshape(4,-1,1)),axis=2).transpose(1,0,2)
    return result`
can solve the problem

**Comment by someoneAlready** [comment]
@MrKamiZhou hi, I use restore_rectangle_v1 to substitute original code, but get different result. Did you get the same result?

**Comment by MrKamiZhou** [comment]
> @MrKamiZhou hi, I use restore_rectangle_v1 to substitute original code, but get different result. Did you get the same result?

no, it will return a slightly different result. Actually, you will get similar result even if you randomly shuffle the score_map...

**Comment by someoneAlready** [comment]
@MrKamiZhou Do you have QQ or Wechat? Maybe we can make a friend.

**Comment by Yangchao0053** [comment]
刚才我也发现这个问题，原来已经有人提过了。 这个错误应该影响不大，所以没有被 发现。
